File: Baekjoon/2606.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [totalNode, edgeNum, ...edges] = fs.readFileSync(filePath).toString().trim().split("\n");
let graph = new Array(+totalNode + 1).fill().map(() => []);
let visited = new Array(+totalNode + 1).fill(0);
let count = 0;

for (i = 0; i < +edgeNum; i++) {
  let edge = edges[i].split(" ");
  let first = +edge[0];
  let last = +edge[1];

  graph[first].push(last);
  graph[last].push(first);
}

visited[1] = 1;

const dfs = (first) => {
  graph[first].forEach((ele) => {
    if (!visited[ele]) {
      visited[ele] = 1;
      count++;
      dfs(ele);
    }
  });
};

dfs(1);

console.log(count);

File: python/Baekjoon/2606.py
n = int(input())
linkedNum = int(input())
graph = [[] for _ in range(n + 1)]
for t in range(linkedNum):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
visited = [False] * (n + 1)

def dfs(graph, v, visited):
    visited[v] = True
    for i in graph[v]:
        if not visited[i]: dfs(graph, i, visited)

dfs(graph, 1, visited)

print(sum(visited) - 1)